great_expectations/execution_engine/sparkdf_execution_engine.py

```diff
@@ -271,7 +271,7 @@ def get_or_create_spark_session(
 
         spark_session: pyspark.SparkSession
         try:
-            spark_session = pyspark.SparkConnectSession.builder.getOrCreate()
+            spark_session = pyspark.SparkConnectSession.builder.getOrCreate()  # type: ignore [assignment]
```

Comment:
not sure why this PR triggered it, but mypy started complaining about this line. Changing the type on 272 to a union including SparkConnectSession just makes results in errors further down. Going with a type assignment because the spark vs spark connect types are a mess and we know this works.